diff -urN a/mediastreamer2/src/Makefile.am b/mediastreamer2/src/Makefile.am
--- a/mediastreamer2/src/Makefile.am	2014-04-24 14:10:10.611593606 +0200
+++ b/mediastreamer2/src/Makefile.am	2014-04-24 14:09:10.814825794 +0200
@@ -373,7 +373,8 @@
 libmediastreamer_voip_la_LIBADD+=       -lole32 \
                                         -loleaut32\
                                         -lwinmm \
-                                        -luuid
+                                        -luuid \
+                                        -lstdc++
 endif
 
 if BUILD_VP8
